refactored update-version.sh to handle new branching strategy

rockhowse opened the pull request:
## Description
This PR supports handling the new main branch strategy outlined below:

* [RSN 47 - Changes to RAPIDS branching strategy in 25.12](https://docs.rapids.ai/notices/rsn0047/)

The `update-version.sh` script should now supports two modes controlled via  `CLI` params or `ENV` vars:

CLI arguments: `--run-context=main|release`
ENV var `RAPIDS_RUN_CONTEXT=main|release`

xref: https://github.com/rapidsai/build-planning/issues/224

bdice commented on cpp/examples/downstream/cmake/get_kvikio.cmake:
```diff
@@ -13,7 +13,7 @@ function(find_and_configure_kvikio MIN_VERSION)
             GIT_REPOSITORY
             https://github.com/rapidsai/kvikio.git
-    GIT_TAG branch-${MIN_VERSION}
+    GIT_TAG release/${MIN_VERSION}
```
I'm happy with using `main` and overwriting with `release/...` in release branches.

rockhowse commented:
Changes after setting GIT_TAG to `main` in the main branch: 

command: 
```bash
./ci/release/update-version.sh --run-context=main 26.02.00
Using run-context from CLI: main
Preparing development branch update 25.12.00a => 26.02.00 (targeting main branch)
```

16 files changed
```bash
git status
On branch ops-4339-update-version-sh-support-main-branching-strategy
Your branch is up to date with 'origin/ops-4339-update-version-sh-support-main-branching-strategy'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .devcontainer/cuda12.9-conda/devcontainer.json
	modified:   .devcontainer/cuda12.9-pip/devcontainer.json
	modified:   .devcontainer/cuda13.0-conda/devcontainer.json
	modified:   .devcontainer/cuda13.0-pip/devcontainer.json
	modified:   .github/workflows/build.yaml
	modified:   .github/workflows/pr.yaml
	modified:   .github/workflows/test.yaml
	modified:   VERSION
	modified:   conda/environments/all_cuda-129_arch-aarch64.yaml
	modified:   conda/environments/all_cuda-129_arch-x86_64.yaml
	modified:   conda/environments/all_cuda-130_arch-aarch64.yaml
	modified:   conda/environments/all_cuda-130_arch-x86_64.yaml
	modified:   cpp/examples/downstream/cmake/get_kvikio.cmake
	modified:   dependencies.yaml
	modified:   java/pom.xml
	modified:   python/kvikio/pyproject.toml
```

diff 
* 
```diff
git diff | cat 
diff --git a/.devcontainer/cuda12.9-conda/devcontainer.json b/.devcontainer/cuda12.9-conda/devcontainer.json
index b220017..c5effd0 100644
--- a/.devcontainer/cuda12.9-conda/devcontainer.json
+++ b/.devcontainer/cuda12.9-conda/devcontainer.json
@@ -5,19 +5,19 @@
     "args": {
       "CUDA": "12.9",
       "PYTHON_PACKAGE_MANAGER": "conda",
-      "BASE": "rapidsai/devcontainers:25.12-cpp-mambaforge"
+      "BASE": "rapidsai/devcontainers:26.02-cpp-mambaforge"
     }
   },
   "runArgs": [
     "--rm",
     "--name",
-    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-25.12-cuda12.9-conda",
+    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-26.02-cuda12.9-conda",
     "--ulimit",
     "nofile=500000"
   ],
   "hostRequirements": {"gpu": "optional"},
   "features": {
-    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:25.12": {}
+    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:26.2": {}
   },
   "overrideFeatureInstallOrder": [
     "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils"
diff --git a/.devcontainer/cuda12.9-pip/devcontainer.json b/.devcontainer/cuda12.9-pip/devcontainer.json
index e15fee7..49702d7 100644
--- a/.devcontainer/cuda12.9-pip/devcontainer.json
+++ b/.devcontainer/cuda12.9-pip/devcontainer.json
@@ -5,19 +5,19 @@
     "args": {
       "CUDA": "12.9",
       "PYTHON_PACKAGE_MANAGER": "pip",
-      "BASE": "rapidsai/devcontainers:25.12-cpp-cuda12.9"
+      "BASE": "rapidsai/devcontainers:26.02-cpp-cuda12.9"
     }
   },
   "runArgs": [
     "--rm",
     "--name",
-    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-25.12-cuda12.9-pip",
+    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-26.02-cuda12.9-pip",
     "--ulimit",
     "nofile=500000"
   ],
   "hostRequirements": {"gpu": "optional"},
   "features": {
-    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:25.12": {}
+    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:26.2": {}
   },
   "overrideFeatureInstallOrder": [
     "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils"
diff --git a/.devcontainer/cuda13.0-conda/devcontainer.json b/.devcontainer/cuda13.0-conda/devcontainer.json
index 4788a99..b1dec3d 100644
--- a/.devcontainer/cuda13.0-conda/devcontainer.json
+++ b/.devcontainer/cuda13.0-conda/devcontainer.json
@@ -5,19 +5,19 @@
     "args": {
       "CUDA": "13.0",
       "PYTHON_PACKAGE_MANAGER": "conda",
-      "BASE": "rapidsai/devcontainers:25.12-cpp-mambaforge"
+      "BASE": "rapidsai/devcontainers:26.02-cpp-mambaforge"
     }
   },
   "runArgs": [
     "--rm",
     "--name",
-    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-25.12-cuda13.0-conda",
+    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-26.02-cuda13.0-conda",
     "--ulimit",
     "nofile=500000"
   ],
   "hostRequirements": {"gpu": "optional"},
   "features": {
-    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:25.12": {}
+    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:26.2": {}
   },
   "overrideFeatureInstallOrder": [
     "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils"
diff --git a/.devcontainer/cuda13.0-pip/devcontainer.json b/.devcontainer/cuda13.0-pip/devcontainer.json
index ff862c5..df77a93 100644
--- a/.devcontainer/cuda13.0-pip/devcontainer.json
+++ b/.devcontainer/cuda13.0-pip/devcontainer.json
@@ -5,19 +5,19 @@
     "args": {
       "CUDA": "13.0",
       "PYTHON_PACKAGE_MANAGER": "pip",
-      "BASE": "rapidsai/devcontainers:25.12-cpp-cuda13.0"
+      "BASE": "rapidsai/devcontainers:26.02-cpp-cuda13.0"
     }
   },
   "runArgs": [
     "--rm",
     "--name",
-    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-25.12-cuda13.0-pip",
+    "${localEnv:USER:anon}-rapids-${localWorkspaceFolderBasename}-26.02-cuda13.0-pip",
     "--ulimit",
     "nofile=500000"
   ],
   "hostRequirements": {"gpu": "optional"},
   "features": {
-    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:25.12": {}
+    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:26.2": {}
   },
   "overrideFeatureInstallOrder": [
     "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils"
diff --git a/.github/workflows/build.yaml b/.github/workflows/build.yaml
index 9a698f5..c287612 100644
--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -69,7 +69,7 @@ jobs:
       arch: "amd64"
       branch: ${{ inputs.branch }}
       build_type: ${{ inputs.build_type || 'branch' }}
-      container_image: "rapidsai/ci-conda:25.12-latest"
+      container_image: "rapidsai/ci-conda:26.02-latest"
       date: ${{ inputs.date }}
       node_type: "gpu-l4-latest-1"
       script: "ci/build_docs.sh"
diff --git a/.github/workflows/pr.yaml b/.github/workflows/pr.yaml
index 1650da8..329f47c 100644
--- a/.github/workflows/pr.yaml
+++ b/.github/workflows/pr.yaml
@@ -112,7 +112,7 @@ jobs:
       build_type: pull-request
       node_type: "gpu-l4-latest-1"
       arch: "amd64"
-      container_image: "rapidsai/ci-conda:25.12-latest"
+      container_image: "rapidsai/ci-conda:26.02-latest"
       script: "ci/test_java.sh"
   conda-python-build:
     needs: conda-cpp-build
@@ -137,7 +137,7 @@ jobs:
       build_type: pull-request
       node_type: "gpu-l4-latest-1"
       arch: "amd64"
-      container_image: "rapidsai/ci-conda:25.12-latest"
+      container_image: "rapidsai/ci-conda:26.02-latest"
       script: "ci/build_docs.sh"
   devcontainer:
     needs: telemetry-setup
diff --git a/.github/workflows/test.yaml b/.github/workflows/test.yaml
index 3f8d231..324a641 100644
--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -51,5 +51,5 @@ jobs:
       sha: ${{ inputs.sha }}
       node_type: "gpu-l4-latest-1"
       arch: "amd64"
-      container_image: "rapidsai/ci-conda:25.12-latest"
+      container_image: "rapidsai/ci-conda:26.02-latest"
       script: "ci/test_java.sh"
diff --git a/VERSION b/VERSION
index 7924af6..5c33046 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-25.12.00
+26.02.00
diff --git a/conda/environments/all_cuda-129_arch-aarch64.yaml b/conda/environments/all_cuda-129_arch-aarch64.yaml
index d2c1dd9..b236b13 100644
--- a/conda/environments/all_cuda-129_arch-aarch64.yaml
+++ b/conda/environments/all_cuda-129_arch-aarch64.yaml
@@ -32,7 +32,7 @@ dependencies:
 - python>=3.10,<3.14
 - rangehttpserver
 - rapids-build-backend>=0.4.0,<0.5.0.dev0
-- rapids-dask-dependency==25.12.*,>=0.0.0a0
+- rapids-dask-dependency==26.2.*,>=0.0.0a0
 - scikit-build-core>=0.10.0
 - sphinx
 - sphinx-click
diff --git a/conda/environments/all_cuda-129_arch-x86_64.yaml b/conda/environments/all_cuda-129_arch-x86_64.yaml
index 9b3292a..b30b803 100644
--- a/conda/environments/all_cuda-129_arch-x86_64.yaml
+++ b/conda/environments/all_cuda-129_arch-x86_64.yaml
@@ -32,7 +32,7 @@ dependencies:
 - python>=3.10,<3.14
 - rangehttpserver
 - rapids-build-backend>=0.4.0,<0.5.0.dev0
-- rapids-dask-dependency==25.12.*,>=0.0.0a0
+- rapids-dask-dependency==26.2.*,>=0.0.0a0
 - scikit-build-core>=0.10.0
 - sphinx
 - sphinx-click
diff --git a/conda/environments/all_cuda-130_arch-aarch64.yaml b/conda/environments/all_cuda-130_arch-aarch64.yaml
index d1db3c2..abba819 100644
--- a/conda/environments/all_cuda-130_arch-aarch64.yaml
+++ b/conda/environments/all_cuda-130_arch-aarch64.yaml
@@ -32,7 +32,7 @@ dependencies:
 - python>=3.10,<3.14
 - rangehttpserver
 - rapids-build-backend>=0.4.0,<0.5.0.dev0
-- rapids-dask-dependency==25.12.*,>=0.0.0a0
+- rapids-dask-dependency==26.2.*,>=0.0.0a0
 - scikit-build-core>=0.10.0
 - sphinx
 - sphinx-click
diff --git a/conda/environments/all_cuda-130_arch-x86_64.yaml b/conda/environments/all_cuda-130_arch-x86_64.yaml
index f37bf67..33cea58 100644
--- a/conda/environments/all_cuda-130_arch-x86_64.yaml
+++ b/conda/environments/all_cuda-130_arch-x86_64.yaml
@@ -32,7 +32,7 @@ dependencies:
 - python>=3.10,<3.14
 - rangehttpserver
 - rapids-build-backend>=0.4.0,<0.5.0.dev0
-- rapids-dask-dependency==25.12.*,>=0.0.0a0
+- rapids-dask-dependency==26.2.*,>=0.0.0a0
 - scikit-build-core>=0.10.0
 - sphinx
 - sphinx-click
diff --git a/cpp/examples/downstream/cmake/get_kvikio.cmake b/cpp/examples/downstream/cmake/get_kvikio.cmake
index 7a257e7..fa61ef3 100644
--- a/cpp/examples/downstream/cmake/get_kvikio.cmake
+++ b/cpp/examples/downstream/cmake/get_kvikio.cmake
@@ -23,4 +23,4 @@ function(find_and_configure_kvikio MIN_VERSION)

 endfunction()

-find_and_configure_kvikio("25.12")
+find_and_configure_kvikio("26.02")
diff --git a/dependencies.yaml b/dependencies.yaml
index 0ab4199..3b08785 100644
--- a/dependencies.yaml
+++ b/dependencies.yaml
@@ -151,7 +151,7 @@ dependencies:
     common:
       - output_types: conda
         packages: &libkvikio_packages
-          - libkvikio==25.12.*,>=0.0.0a0
+          - libkvikio==26.2.*,>=0.0.0a0
     specific:
       - output_types: [requirements, pyproject]
         matrices:
@@ -159,12 +159,12 @@ dependencies:
               cuda: "12.*"
               cuda_suffixed: "true"
             packages:
-              - libkvikio-cu12==25.12.*,>=0.0.0a0
+              - libkvikio-cu12==26.2.*,>=0.0.0a0
           - matrix:
               cuda: "13.*"
               cuda_suffixed: "true"
             packages:
-              - libkvikio-cu13==25.12.*,>=0.0.0a0
+              - libkvikio-cu13==26.2.*,>=0.0.0a0
           - {matrix: null, packages: *libkvikio_packages}
   build-py-wrapper:
     common:
@@ -233,7 +233,7 @@ dependencies:
     common:
       - output_types: conda
         packages:
-          - &libkvikio_unsuffixed libkvikio==25.12.*,>=0.0.0a0
+          - &libkvikio_unsuffixed libkvikio==26.2.*,>=0.0.0a0
       - output_types: requirements
         packages:
           # pip recognizes the index as a global option for the requirements.txt file
@@ -247,12 +247,12 @@ dependencies:
               cuda: "12.*"
               cuda_suffixed: "true"
             packages:
-              - libkvikio-cu12==25.12.*,>=0.0.0a0
+              - libkvikio-cu12==26.2.*,>=0.0.0a0
           - matrix:
               cuda: "13.*"
               cuda_suffixed: "true"
             packages:
-              - libkvikio-cu13==25.12.*,>=0.0.0a0
+              - libkvikio-cu13==26.2.*,>=0.0.0a0
           - {matrix: null, packages: [*libkvikio_unsuffixed]}
   docs:
     common:
@@ -315,14 +315,14 @@ dependencies:
     common:
       - output_types: conda
         packages:
-          - libkvikio==25.12.*,>=0.0.0a0
-          - libkvikio-tests==25.12.*,>=0.0.0a0
+          - libkvikio==26.2.*,>=0.0.0a0
+          - libkvikio-tests==26.2.*,>=0.0.0a0
   test_kvikio:
     common:
       - output_types: conda
         packages:
-          - libkvikio==25.12.*,>=0.0.0a0
-          - kvikio==25.12.*,>=0.0.0a0
+          - libkvikio==26.2.*,>=0.0.0a0
+          - kvikio==26.2.*,>=0.0.0a0
   test_cpp:
     common:
       - output_types: conda
@@ -332,7 +332,7 @@ dependencies:
     common:
       - output_types: [conda, requirements, pyproject]
         packages:
-          - rapids-dask-dependency==25.12.*,>=0.0.0a0
+          - rapids-dask-dependency==26.2.*,>=0.0.0a0
           - pytest
           - pytest-asyncio
           - pytest-cov
diff --git a/java/pom.xml b/java/pom.xml
index 1dc011b..be956c8 100644
--- a/java/pom.xml
+++ b/java/pom.xml
@@ -10,7 +10,7 @@ SPDX-License-Identifier: Apache-2.0

   <groupId>ai.rapids.kvikio</groupId>
   <artifactId>cufile</artifactId>
-  <version>25.12.0-SNAPSHOT</version>
+  <version>26.02.0-SNAPSHOT</version>

   <name>cuFile</name>
   <description>
diff --git a/python/kvikio/pyproject.toml b/python/kvikio/pyproject.toml
index 2a2d4c4..2d24cb9 100644
--- a/python/kvikio/pyproject.toml
+++ b/python/kvikio/pyproject.toml
@@ -20,7 +20,7 @@ license = { text = "Apache-2.0" }
 requires-python = ">=3.10"
 dependencies = [
     "cupy-cuda13x>=13.6.0",
-    "libkvikio==25.12.*,>=0.0.0a0",
+    "libkvikio==26.2.*,>=0.0.0a0",
     "numpy>=1.23,<3.0a0",
     "packaging",
 ] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
@@ -46,7 +46,7 @@ test = [
     "pytest-cov",
     "pytest-timeout",
     "rangehttpserver",
-    "rapids-dask-dependency==25.12.*,>=0.0.0a0",
+    "rapids-dask-dependency==26.2.*,>=0.0.0a0",
     "zarr>=3.0.0,<4.0.0; python_version >= '3.11'",
 ] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
 zarr = [
@@ -99,7 +99,7 @@ matrix-entry = "cuda_suffixed=true;use_cuda_wheels=true"
 requires = [
     "cmake>=3.30.4",
     "cython>=3.0.0",
-    "libkvikio==25.12.*,>=0.0.0a0",
+    "libkvikio==26.2.*,>=0.0.0a0",
     "ninja",
 ] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
 ```

rockhowse commented:
/merge
